dhcpv6/dhcpv6message.go

```diff
@@ -65,49 +65,50 @@ func GetTime() uint32 {
 	return uint32((now.Nanoseconds() / 1000000000) % 0xffffffff)
 }
 
-// NewSolicitForInterface creates a new SOLICIT message with DUID-LLT, using the
-// given network interface's hardware address and current time
-func NewSolicitForInterface(ifname string, modifiers ...Modifier) (DHCPv6, error) {
+// NewSolicitWithCID creates a new SOLICIT message with CID.
+func NewSolicitWithCID(duid Duid, modifiers ...Modifier) (DHCPv6, error) {
 	d, err := NewMessage()
 	if err != nil {
 		return nil, err
 	}
 	d.(*DHCPv6Message).SetMessage(SOLICIT)
-	iface, err := net.InterfaceByName(ifname)
-	if err != nil {
-		return nil, err
-	}
-	cid := OptClientId{
-		Cid: Duid{
-			Type:          DUID_LLT,
-			HwType:        iana.HwTypeEthernet,
-			Time:          GetTime(),
-			LinkLayerAddr: iface.HardwareAddr,
-		},
-	}
-
-	d.AddOption(&cid)
-	oro := OptRequestedOption{}
+	d.AddOption(&OptClientId{Cid: duid})
+	oro := new(OptRequestedOption)
 	oro.SetRequestedOptions([]OptionCode{
 		DNS_RECURSIVE_NAME_SERVER,
 		DOMAIN_SEARCH_LIST,
 	})
-	d.AddOption(&oro)
-	d.AddOption(&OptElapsedTime{})
+	d.AddOption(oro)
+	d.AddOption(new(OptElapsedTime))
 	// FIXME use real values for IA_NA
-	iaNa := OptIANA{}
+	iaNa := new(OptIANA)
```

Comment:
note: there is no advantage or disadvantage in using `new` vs `&OptIana{}` here - in this specific case it's just style preference. This one can go, but in general I prefer we don't make unnecessary style changes